crate/operator/grand_central.py

```diff
@@ -390,6 +393,33 @@ async def create_grand_central_backend(
         core = CoreV1Api(api_client)
         networking = NetworkingV1Api(api_client)
 
+        await call_kubeapi(
+            core.create_namespaced_secret,
+            logger,
+            continue_on_conflict=True,
+            namespace=namespace,
+            body=get_gc_user_secret(owner_references, name, labels),
+        )
+
+        # circular import issue
+        from crate.operator.operations import get_cratedb_resource
+
+        # Before trying to create the gc_admin user, we need to make sure
+        # CrateDB is already started and the system user exists.
```

Comment:
Why do we care? Won't it fail and retry?

We might want to consider splitting this out into a subhandler, so that it wouldn't block creating the deployment, service etc. Cause right now it will effectively delay creating grand central until the cluster finishes bootstrapping.

Reply:
In general, all handlers in kopf should be idempotent, and should be able to retry if an error happens half way through (so i.e., if you manage to create the user but then creating the deployment fails, it would retry the entire handler - and you don't want creating the gc_admin user to fail if it already exists).

So I think it might be a good idea to do this creation in a subhandler instead. And possibly also move the deployment/service/ingress creation into one too...

Reply:
Makes sense. I will move the user creation into a subhandler and create a follow up PR for deployment/service/ingress/...